File: Frontend/AZBJ_OTC_QC_FMB/src/app/services/image-details.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { ImageDetails } from '../shared/models/image-details.model';

@Injectable({
  providedIn: 'root'
})
export class ImageDetailsService {
  private apiUrl = 'http://your-api-url.com/api/image-details';

  constructor(private http: HttpClient) {}

  getImageDetails(): Observable<ImageDetails> {
    return this.http.get<ImageDetails>(`${this.apiUrl}`);
  }

  logTransferDetails(contract_id: string, appln_no: string, log_message: string): Observable<void> {
    return this.http.post<void>(`${this.apiUrl}/log-transfer`, { contract_id, appln_no, log_message });
  }

  logErrorDetails(contract_id: string, appln_no: string, log_message: string): Observable<void> {
    return this.http.post<void>(`${this.apiUrl}/log-error`, { contract_id, appln_no, log_message });
  }
}

File: Backend/AZBJ_OTC_QC_FMB/my-spring-boot-app/src/main/java/com/example/service/ImageDetailsService.java
package com.example.service;

import com.example.dao.ImageDetailsDao;
import com.example.model.ImageDetails;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

@Service
public class ImageDetailsService {

    @Autowired
    private ImageDetailsDao imageDetailsDao;

    public ImageDetails getImageDetails() {
        return imageDetailsDao.getImageDetails();
    }

    public void logTransferDetails(String contract_id, String appln_no, String log_message) {
        imageDetailsDao.logTransferDetails(contract_id, appln_no, log_message);
    }

    public void logErrorDetails(String contract_id, String appln_no, String log_message) {
        imageDetailsDao.logErrorDetails(contract_id, appln_no, log_message);
    }
}
